tests/topotests/ospf6-topo1-vrf/test_ospf6_topo1_vrf.py

```diff
@@ -0,0 +1,505 @@
+
+            isConverged = False
+
+            for line in resStr.splitlines():
+                res1 = pat1.match(line)
+                if res1:
+                    isConverged = True
+                    res2 = pat2.search(line)
+
+                    if res2 == None:
+                        isConverged = False
+                        break
+
+            if isConverged == False:
+                logger.info("Waiting for {}".format(router))
+                sys.stdout.flush()
+                break
+
+        if isConverged:
+            logger.info("Done")
+            break
+        else:
+            sleep(5)
+            timeout -= 5
+
+    if timeout == 0:
+        # Bail out with error if a router fails to converge
+        ospfStatus = rnode.vtysh_cmd("show ipv6 ospf neigh")
+        assert False, "OSPFv6 did not converge:\n{}".format(ospfStatus)
+
+    logger.info("OSPFv3 converged.")
+
+    # For debugging, uncomment the next line
+    # tgen.mininet_cli()
+
+    # Make sure that all daemons are still running
+    if tgen.routers_have_failure():
+        assert tgen.errors == "", tgen.errors
+
+
+def compare_show_ipv6_vrf(rname, expected):
+    """
+    Calls 'show ipv6 route' for router `rname` and compare the obtained
+    result with the expected output.
+    """
+    tgen = get_topogen()
+
+    # Use the vtysh output, with some masking to make comparison easy
+    vrf_name = "{0}-cust1".format(rname)
+    current = topotest.ip6_route_zebra(tgen.gears[rname], vrf_name)
+    
+    # Use just the 'O'spf lines of the output
+    linearr = []
+    for line in current.splitlines():
+        if re.match("^O", line):
+            linearr.append(line)
+
+    current = "\n".join(linearr)
+
+    return topotest.difflines(
+        topotest.normalize_text(current),
+        topotest.normalize_text(expected),
+        title1="Current output",
+        title2="Expected output",
+    )
+
+
+def test_ospfv3_routingTable():
+
+    tgen = get_topogen()
+    if tgen.routers_have_failure():
+        pytest.skip("skipped because of router(s) failure")
+
+    # For debugging, uncomment the next line
+    # tgen.mininet_cli()
+    # Verify OSPFv3 Routing Table
+    for router, rnode in tgen.routers().iteritems():
+        logger.info('Waiting for router "%s" convergence', router)
+
+        # Load expected results from the command
+        reffile = os.path.join(CWD, "{}/show_ipv6_vrf_route.ref".format(router))
+        expected = open(reffile).read()
+
+        # Run test function until we get an result. Wait at most 60 seconds.
+        test_func = partial(compare_show_ipv6_vrf, router, expected)
+        result, diff = topotest.run_and_expect(test_func, "", count=120, wait=0.5)
+        assert result, "OSPFv3 did not converge on {}:\n{}".format(router, diff)
+
+
+def test_linux_ipv6_kernel_routingTable():
+
+    dist = platform.dist()
+
+    if dist[1] == "16.04":
```

Comment:
Please lets not write code using this anymore. The PR #8253 adds the tools to fix this.

`platform.dist` no longer exists in Python3 (it moved to a package)

Reply:
Rafael, do you mean it is better to use iproute version than kernel release?

Reply:
> Please lets not write code using this anymore. The PR #8253 adds the tools to fix this.
> 
> `platform.dist` no longer exists in Python3 (it moved to a package)

Ok, rafael will change it.

Reply:
Resolved